modules/web/src/app/shared/components/labels/component.ts

```diff
@@ -12,20 +12,36 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import {Component, Input, OnChanges, OnInit, SimpleChanges} from '@angular/core';
+import {AfterViewInit, ChangeDetectorRef, Component, Input, OnChanges, OnInit, SimpleChanges} from '@angular/core';
 
 @Component({
   selector: 'km-labels',
   templateUrl: './template.html',
+  styleUrls: ['style.scss'],
 })
-export class LabelsComponent implements OnInit, OnChanges {
+export class LabelsComponent implements OnInit, OnChanges, AfterViewInit {
   @Input() labels = {};
   @Input() limit: number;
   @Input() emptyMessage = '';
+  @Input() oneLineLimit = false;
   labelKeys: string[] = [];
+  showHiddenLabels = false;
+  showAndHideExtraLabels = false;
+
+  get showLabelsText(): string {
+    return this.showHiddenLabels ? 'SHOW ALL' : 'HIDE ALL';
+  }
+
+  constructor(private readonly _cdr: ChangeDetectorRef) {}
 
   ngOnInit(): void {
     this._updateLabelKeys();
+    this.oneLineLimit && this._checkLabelsHeight();
```

Comment:
`ngAfterViewInit` will run after `ngOnInit` so I think we can remove this line here and keep it in `ngAfterViewInit`.